apicast/src/proxy.lua

```diff
@@ -281,7 +281,7 @@ function _M:call(host)
   self:set_backend_upstream(service)
 
   if service.backend_version == 'oauth' then
-    local f, params = oauth.call()
+    local f, params = oauth.call(_, _, service)
```

Comment:
this reveals some bad interface
if those two parameters are not needed or used there should be some other way to initialize this object, maybe change the order and let service be first ?